hooks/post_gen_project.py

```diff
@@ -26,19 +26,31 @@ def invoke_shell(command):
         print(e.output)
         raise e
     print(decode_string(output))
+    if return_output:
+            return decode_string(output)
 
 
 def git_init_and_tag():
     """Invoke the initial git and tag with 0.0.0 to make an initial version for Versioneer to ID"""
-    # Initialize git
-    invoke_shell("git init")
-    # Add files
+    
+    # Check if we are in a git repository
+    directory_status = invoke_shell("git status", return_output=True)
+    # Create a repository if not already in one.
+    if 'fatal' in directory_status:
+        # Initialize git
+        invoke_shell("git init")
+
+    # Add files created by cookiecutter 
     invoke_shell("git add .")
     invoke_shell(
```

Comment:
If we do not do a `git init` we probably do not want to commit as this is likely a destructive process on the repository. That is, it will overwrite current code in the repo and the user will need to verify and commit this themselves. We should likely print out a warning message to this effect.

Reply:
It is actually not destructive to the repository. This script runs inside of the folder that cookie cutter creates, meaning that when you do “git add .”, it only adds and commits contents inside the template project folder cookiecutter has created. This is the same as previous, except that a new sub module is not created with a git init. I think it is rarely the case that someone wants to create a git repository inside a git repository, and that is what the cookie cutter does without this check.

I tested it against a repository which had uncommitted changes, and these changes were all preserved with running the cookiecutter with this check. Of course, automated testing could be written to ensure this as well.

Reply:
I see so you are running these within an existing repo expecting it to build out a new folder. In that case we definitely do not want to commit the folder as whoever runs this will then need to move the generated files to their correct locations.

Reply:
I don't quite understand because this is how the CookieCutter worked before this pull request. It has always built out a new folder. The issue is that people do run it within existing repositories, and a git init in existing repositories causes a lot of chaos. The only difference between what it does now and what it did before is an 'if' statement to check if it's in a repository, and only inits if it is not.

There can be a commit for the initial structure of the repository, then the user can create other commits when they move files.

Reply:
If this is run within an existing repo, at minimum, we do not want to do `git add .`, which will add everything, and likely not be what the user wanted. 

The question then becomes what do we add? One stance is as @dgasmith said is to add nothing and expect the user to move the files and add them manually. This is the safest option and likely the one we should go with.

However, there is another option, because we know exactly what files will be created from the cookiecutter, we could opt to add only those files, it will take some engineering, but could be done and it would ensure we can get a commit which ensures the "Created by MolSSI cookiecutter" tag is somewhere so we can loosely track the project usage. 

In either option, if this is being created in an existing git repository, then we do not want to call the `git tag` command

Reply:
I thought of this. Since it is in the folder that cookiecutter creates when the hit add runs, “git add .” only adds files which cookiecutter creates. I tested it.

Reply:
Cookiecutter changes directories to be in the project folder before committing.

Reply:
I will note that per your second proposal 

>>> However, there is another option, because we know exactly what files will be created from the cookiecutter, we could opt to add only those files, it will take some engineering, but could be done and it would ensure we can get a commit which ensures the "Created by MolSSI cookiecutter" tag is somewhere so we can loosely track the project usage.

The CookieCutter *by nature* can only commit what it creates because it cds into the directory it creates. My thought was to keep the commit so the tag would be present in the commit history. The PR for the CookieCutter now behaves the way described in the above paragraph.